infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerService.java

```diff
@@ -98,18 +97,26 @@ private void handle(BrokerStatusChangedEvent event) {
       LOG.error("Broker event skipped due to illegal content: {}", event);
       return;
     }
-    eventService.publish(new BrokerEvent(event, parseTooling(encodedTooling)));
-  }
-
-  @Nullable
-  private List<ChePlugin> parseTooling(String toolingString) {
-    if (!isNullOrEmpty(toolingString)) {
+    List<ChePlugin> tooling = null;
+    if (isNullOrEmpty(event.getError()) && !isNullOrEmpty(encodedTooling)) {
       try {
-        return objectMapper.readValue(toolingString, new TypeReference<List<ChePlugin>>() {});
+        tooling = parseTooling(encodedTooling);
       } catch (IOException e) {
-        LOG.error("Parsing Che plugin broker event failed. Error: " + e.getMessage(), e);
+        if (isNullOrEmpty(event.getError())) {
+          event.withError(e.getMessage());
+        }
       }
     }
-    return null;
+    eventService.publish(new BrokerEvent(event, tooling));
```

Comment:
Event with null tooling will be published if `encodedTooling` is null or empty, right? If yes, please handle it.

Reply:
Yes, it will be sent.  Can you elaborate on what needs to be handled?

Reply:
I guess because `BrokerEvent` supposed to contain tooling or error. But I may be wrong.

Reply:
You are right but I'm not following what needs to be changed